DIPs/DIP1NNN-MVF.md

```diff
@@ -0,0 +1,159 @@
+  * Templates without an explicit `@safe`, `@trusted`, or `@system` attribute will remain agnostic, inferring their attributes from the site of their insantiation, regardless of whether the compiler was invoked with one of the aforementioned compiler flag or not, maintaining the status quo.
+
+DMD will also be updated with a new `version` identifier `D_SystemByDefault` that will be set to `true` any time the compiler is invoked with `-revert=safeByDefault` or neither flag.  Utilizing D's fantastic design by introspection features, users will be able to use `D_SystemByDefault` in their code to futher manage the transition with greater detail and greater precision, as illustrated below.
+
+```D
+// Ensuring a module is only compilable with a `@safe`-by-default compiler
+module myModule;
+
+version (D_SystemByDefault)
+{
+    static assert(false, "This module should only be compiled with a `@safe`-by-default compiler.  Use `-transtion=safeByDefault`");
+}
+
+// Module implementation
+```
+
+```D
+// Ensuring a module is only compilable with a `@system`-by-default compiler
+module myModule;
+
+version (D_SystemByDefault) { } else
+{
+    static assert(false, "This module should only be compiled with a `@system`-by-default compiler.  Use `-revert=safeByDefault`");
+}
+
+// Module implementation
+```
+
+A changelog entry detailing said features will accompany the release in which they appear.
+
+### Stage 3: Epoc - D Becomes `@safe`-by-Default Programming Language
+
+DMD will be updated to be a `@safe`-by-default compiler.
+  * Invoking the compiler with neither the `-transition=safeByDefault` flag nor the `-revert=safeByDefault` flag will cause the compiler to produce code that is `@safe`-by-default.  The deprecation warning message added in Stage 2 will be removed.
+  * Invoking the compiler with `-transtion=safeByDefault` have no effect on code, but will emit a deprecation warning stating that the `-trasition=safeByDefault` flag is no longer needed and should no longer be used.
+  * Invoking the compiler with `-revert=safeByDefault` will cause the compiler to produce `@system` code by default, allowing the user to opt out of the new `@safe`-by-default behavior, and revert to the old behavior.
+  * Templates without an explicit `@safe`, `@trusted`, or `@system` attribute will remain agnostic, inferring their attributes from the site of their insantiation, regardless of whether the compiler was invoked with a `-transition` or `-revert` flag or not, maintaining the status quo.
+
+A changelog entry detailing these changes will accompany the release in which they appear.
+
+### Stage 4: Epoc plus 1 year: Deprecate Transition Features
```

Comment:
"Deprecate" is confusing here, because you are actually emitting errors. I would leave the deprecation for 2 years instead of 1. This is a major language update.

Reply:
> "Deprecate" is confusing here, because you are actually emitting errors.

Changed to "invalidate".  I'm still considering the 2 year time;  I haven't decided yet, and my just leave it as an option at approval time.